### PR TITLE
Allow easy plugin of featureflags into E2E

### PR DESCRIPTION
**Pull request by 0xKiwi**
This PR changes the E2E and featureconfig to allow easy E2E testing of featureflags.

### CODE REVIEW

File: shared/featureconfig/flags.go [hunk 1]
```diff
@@ -232,3 +238,15 @@ var BeaconChainFlags = append(deprecatedFlags, []cli.Flag{
...
+// E2EBeaconChainFlags contains a list of the beacon chain feature flags to be tested in E2E.
+var E2EBeaconChainFlags = []string{
+	"--enable-ssz-cache",
+	"--enable-attestation-cache",
+	"--cache-proposer-indices",
+	"--cache-filtered-block-tree",
+	"--enable-skip-slots-cache",
+	"--enable-eth1-data-vote-cache",
+	"--proto-array-forkchoice",
+}
```
Comment (terencechain) [inline shared/featureconfig/flags.go]:
```suggestion
}

```

File: endtoend/README.md [hunk 2]
```diff
@@ -10,8 +11,8 @@ In order to "evaluate" the state of the beacon chain while the E2E is running, t
 Evaluators have 3 parts, the name for it's test name, a `policy` which declares which epoch(s) the evaluator should run, and then the `evaluation` which uses the beacon chain API to determine if the beacon chain passes certain conditions like finality.
 
 ## Current end-to-end tests
-* Minimal Config - 4 beacon nodes, 64 validators, running for 5 epochs
-* Demo Config - 4 beacon nodes, 64 validators, running for 5 epochs
+* Minimal Config - 4 beacon nodes, 64 validators, running for 6 epochs
+* ~~Demo Config - 2 beacon nodes, 16,384 validators, running for 5 epochs~~ Disabled for now
```
Comment (terencechain) [inline endtoend/README.md]: Any reasons for the `~~`??

```suggestion
* Demo Config - 2 beacon nodes, 16,384 validators, running for 5 epochs~~ Disabled for now
```

Comment (terencechain) [reply endtoend/README.md]: Nvm it's the tilting the font

Comment (0xKiwi) [reply endtoend/README.md]: Just figured to scratch it out since it's disabled at the moment.

File: shared/featureconfig/flags.go (hunk 1 above)
Comment (0xKiwi) [reply shared/featureconfig/flags.go]: Fixed 👍

**Pull request merged**